packages/marshal/src/helpers/passStyle-helpers.js

```diff
@@ -110,6 +111,7 @@ harden(getTag);
  */
 export const checkTagRecord = (tagRecord, passStyle, check = x => x) => {
   return (
+    check(isFrozen(tagRecord), X`A tagRecord must be frozen: ${tagRecord}`) &&
     check(
       typeof tagRecord === 'object',
       X`A non-object cannot be a tagRecord: ${tagRecord}`,
```

Comment:
These checks feel like they're in the wrong order; I think we should only check frozen status after confirming that we've got an object. It would probably also make sense to move the `null` check up, and possibly combine it with `typeof` (e.g., ```check(typeof tagRecord === 'object' && tagRecord !== null, X`A non-object cannot be a tagRecord: ${tagRecord}`)```)

Reply:
Nice improvements, thanks. Done.